compiler/src/main/proto/spine/protodata/ast.proto

```diff
@@ -0,0 +1,256 @@
+    string value = 1;
+}
+
+// A Protobuf source file.
+//
+message File {
+
+    // Path to the file in a file system.
+    Path path = 1;
+
+    // Name of the Protobuf package to which the declarations of the file belong.
+    string package_name = 2;
+
+    // The version of the Protobuf syntax.
+    SyntaxVersion syntax = 3;
+
+    // File-level options.
+    repeated Option option = 4;
+
+    // The version of the Protobuf syntax, as defined in a Protobuf source file.
+    //
+    enum SyntaxVersion {
+
+        LV_UNDEFINED = 0;
+
+        PROTO2 = 1;
+
+        PROTO3 = 2;
+    }
+}
+
+// Name of a Protobuf type.
+message TypeName {
+
+    // The name, including the package.
+    string value = 1;
+}
+
+// A Protobuf message type.
+message MessageType {
+
+    TypeName name = 1;
+
+    // The prefix of the type's URL.
+    //
+    // A type URL has the following structure: "<type URL prefix>/<type FQN>". For more info, see
+    // the definition of `google.protobuf.Any.type_url`.
+    //
+    string type_url_prefix = 2;
+
+    // The fields of this type.
+    //
+    // Does not include the fields belonging to a field group.
+    //
+    repeated Field field = 3;
+
+    // The field groups defined in this type.
+    repeated Group group = 4;
+
+    // Message-level options.
+    repeated Option option = 5;
+}
+
+// A primitive Protobuf type.
+enum PrimitiveType {
+
+    PT_UNKNOWN = 0;
+
+    TYPE_DOUBLE = 1;
+    TYPE_FLOAT = 2;
+    TYPE_INT64 = 3;
+    TYPE_UINT64 = 4;
+    TYPE_INT32 = 5;
+    TYPE_FIXED64 = 6;
+    TYPE_FIXED32 = 7;
+    TYPE_BOOL = 8;
+    TYPE_STRING = 9;
+    TYPE_BYTES = 10;
+    TYPE_UINT32 = 11;
+    TYPE_SFIXED32 = 12;
+    TYPE_SFIXED64 = 13;
+    TYPE_SINT32 = 14;
+    TYPE_SINT64 = 15;
+}
+
+// A Protobuf type.
+message Type {
+
+    oneof kind {
+
+        MessageType message = 1;
+
+        PrimitiveType primitive = 2;
+    }
+}
+
+// A name of a field in a message.
+message FieldName {
+
+    // Short name of the field.
+    string value = 1;
+}
+
+// A Protobuf message field.
+message Field {
+
+    FieldName name = 1;
+
+    // The type of the field.
+    Type type = 2;
+
+    // The type in which the field is declared.
+    MessageType declaring_type = 3;
+
+    // The field number.
+    //
+    // Not to be confused with the field's order to declaration.
+    //
+    // For example, in a message:
+    // ```
+    // message Foo {
+    //     string bar = 2;
+    //     int32 baz = 1;
+    // }
+    // ```
+    // Field `bar` has the number 2 and the field `baz` has the number 1.
+    //
+    uint32 number = 4;
+
+    // The field's order to declaration.
+    //
+    // Not to be confused with the field number.
+    //
+    // For example, in a message:
+    // ```
+    // message Foo {
+    //     string bar = 2;
+    //     int32 baz = 1;
+    // }
+    // ```
+    // Field `bar` has the order 0 and the field `baz` has the order 1.
+    //
+    uint32 order_of_declaration =54;
```

Comment:
Please format this one. And, probably, reconsider the `54` (I would understand if it was `42`, but `54` seem misleading).